### PR TITLE
[BEAM-3028] Fixes a bug in DatastoreIO query splitting.

### PR DESCRIPTION
**Pull request by chamikaramj**
We were returning original query instead of the sub-queries resulting in data duplication when reading.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @jkff

**Comment by jkff** [comment]
Wow. Can you add a unit test? Modulo that, LGTM.

**Comment by chamikaramj** [comment]
Added a unit test. Thanks.

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java [hunk 1]
```diff
@@ -628,6 +629,27 @@ public void testSplitQueryFnWithNumSplits() throws Exception {
   }
 
   /**
+   * Tests {@link SplitQueryFn} to make sure that sub-queries are different from the original query
```
Comment (jkff) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java]: I think this should be folded into the previous test, and the function `splitQuery` should be changed to generate unequal queries.

Comment (chamikaramj) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java]: Done.

**Comment by jkff** [comment]
LGTM.

**Comment by chamikaramj** [comment]
Thanks. Will merge after Jenkins tests pass.

**Pull request closed without merging**